test/js/bun/ffi/cc.test.ts

```diff
@@ -14,3 +16,106 @@ it.todoIf(isWindows)("can run a .c file", () => {
 
   expect(result.exitCode).toBe(0);
 });
+
+describe("given an add(a, b) function", () => {
+  const source = /* c */ `
+      int add(int a, int b) {
+        return a + b;
+      }
+    `;
+  let dir;
+
+  beforeAll(() => {
+    dir = tempDirWithFiles("bun-ffi-cc-test", {
+      "add.c": source,
+    });
+  });
+
+  afterAll(async () => {
+    await fs.rm(dir, { recursive: true, force: true });
+  });
+
+  describe("when compiled", () => {
+    let res: Library<{ add: { args: ["int", "int"]; returns: "int" } }>;
+
+    beforeAll(() => {
+      res = cc({
+        source: path.join(dir, "add.c"),
+        symbols: {
+          add: {
+            returns: "int",
+            args: ["int", "int"],
+          },
+        },
+      });
+    });
+
+    afterAll(() => {
+      res.close();
+    });
+
+    it("provides an add symbol", () => {
+      expect(res.symbols.add(1, 2)).toBe(3);
+    });
+
+    it("Only contains 1 symbol", () => {
+      expect(Object.keys(res.symbols)).toHaveLength(1);
+    });
+  }); // </when compiled>
+
+  it("when compiled with a symbol that doesn't exist, throws an error", () => {
+    expect(() => {
+      cc({
+        source: path.join(dir, "add.c"),
+        symbols: { subtract: { args: ["int", "int"], returns: "int" } },
+      });
+    }).toThrow(/"subtract" is missing/);
+  });
+
+  it("when compiled with a `void` return type instead of `int`, returns undefined", () => {
```

Comment:
this is undefined behavior

Reply:
instead another function should be used, like `add`

but tinycc will probably compile it into nothing